src/room/creeps/roleManagers/commune/builder.ts

```diff
@@ -39,7 +39,7 @@ Builder.prototype.getEnergy = function () {
 
     // If there are no sourceHarvesters in the room, harvest a source
 
-    if (!(room.myCreeps.source1Harvester.length + room.myCreeps.source2Harvester.length)) {
+    if (!((room.myCreeps.source1Harvester?.length) || 0 + (room.myCreeps.source2Harvester?.length) || 0)) {
```

Comment:
I get what your doing but is there an change that one of then is zero if yes then keep it otherwise maybe only do for source2